File: data/EXP01/golang_no_system_var/2186/2186.output_2.go
func minSteps(s string, t string) int {
    count := make(map[byte]int)

    for i := 0; i < len(s); i++ {
        count[s[i]]++
    }

    steps := 0

    for i := 0; i < len(t); i++ {
        if count[t[i]] > 0 {
            count[t[i]]--
        } else {
            steps++
        }
    }

    return steps
}

File: data/EXP01/python/2186/2186.output_2.py
# @lc app=leetcode id=2186 lang=python3
class Solution:
    def minSteps(self, s: str, t: str) -> int:
        dict_s = {}
        dict_t = {}
        
        for c in s:
            dict_s[c] = dict_s.get(c, 0) + 1
        
        for c in t:
            dict_t[c] = dict_t.get(c, 0) + 1
        
        steps = 0
        
        for c in s:
            if c not in t:
                steps += 1
            elif dict_s[c] > dict_t[c]:
                steps += dict_s[c] - dict_t[c]
        
        return steps